cpp/src/groupby/sort/group_scan_util.cuh

```diff
@@ -36,27 +42,55 @@
 namespace cudf {
 namespace groupby {
 namespace detail {
+// Error case when no other overload or specialization is available
+template <aggregation::Kind K, typename T, typename Enable = void>
+struct group_scan_functor {
+  template <typename... Args>
+  static std::unique_ptr<column> invoke(Args&&...)
+  {
+    CUDF_FAIL("Unsupported groupby scan type-agg combination.");
+  }
+};
+
 template <aggregation::Kind K>
-struct scan_functor {
+struct group_scan_dispatcher {
   template <typename T>
-  static constexpr bool is_supported()
+  std::unique_ptr<column> operator()(column_view const& values,
+                                     size_type num_groups,
+                                     cudf::device_span<cudf::size_type const> group_labels,
+                                     rmm::cuda_stream_view stream,
+                                     rmm::mr::device_memory_resource* mr)
   {
-    if (K == aggregation::SUM)
-      return cudf::is_numeric<T>() || cudf::is_duration<T>() || cudf::is_fixed_point<T>();
-    else if (K == aggregation::MIN or K == aggregation::MAX)
-      return !cudf::is_dictionary<T>() and is_relationally_comparable<T, T>();
-    else
-      return false;
+    return group_scan_functor<K, T>::invoke(values, num_groups, group_labels, stream, mr);
   }
+};
 
-  template <typename T>
-  std::enable_if_t<is_supported<T>() and not std::is_same_v<T, cudf::string_view>,
-                   std::unique_ptr<column>>
-  operator()(column_view const& values,
-             size_type num_groups,
-             cudf::device_span<cudf::size_type const> group_labels,
-             rmm::cuda_stream_view stream,
-             rmm::mr::device_memory_resource* mr)
+/**
+ * @brief Check if the given aggregation K with data type T is supported in groupby scan.
+ */
+template <aggregation::Kind K, typename T>
+static constexpr bool is_group_scan_supported()
+{
+  if (K == aggregation::SUM)
+    return cudf::is_numeric<T>() || cudf::is_duration<T>() || cudf::is_fixed_point<T>();
+  else if (K == aggregation::MIN or K == aggregation::MAX)
+    return not cudf::is_dictionary<T>() and
+           (is_relationally_comparable<T, T>() or std::is_same_v<T, cudf::struct_view>);
+  else
+    return false;
+}
+
+template <aggregation::Kind K, typename T>
+struct group_scan_functor<
+  K,
+  T,
+  std::enable_if_t<is_group_scan_supported<K, T>() and not std::is_same_v<T, cudf::string_view> and
+                   not std::is_same_v<T, cudf::struct_view>>> {
```

Comment:
The beauty of the dispatch to invoke pattern is that you don't need all of the "not" statements in your `enable_if` anymore. You can just write the condition that _enables_ a particular code path without needing to include the logical conjugation of all other code path conditionals. 
```suggestion
template <aggregation::Kind K, typename T>
struct group_scan_functor<
  K,
  T,
  std::enable_if_t<is_group_scan_supported<K, T>()> {
```

Reply:
See my example from a while back: https://godbolt.org/z/GYd1c47fP